File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAdjacentDistance = function(nums) {
    let n = nums.length;
    let ans = Math.abs(nums[0] - nums[n - 1]);
    for (let i = 1; i < n; i++) {
        let val = Math.abs(nums[i - 1] - nums[i]);
        ans = Math.max(ans, val);
    }
    return ans;
};

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.py
class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        n = len(nums)
        ans = abs(nums[0] - nums[n - 1])
        for i in range(1, n):
            val = abs(nums[i - 1] - nums[i])
            ans = max(ans, val)
        
        return ans